File: downloader/src/main/java/de/viktorlevin/voicedownloader/dto/ExamplesDto.java
package de.viktorlevin.voicedownloader.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ExamplesDto {
    private List<Result> result = new ArrayList<>();

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Result {
        private String text;
        private List<Example> examples = new ArrayList<>();
    }

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Example {
        private String src;
        private String dst;
        private Reference ref;
    }

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Reference {
        private String title;
        private String type;
    }
}

File: downloader/src/main/java/de/viktorlevin/voicedownloader/dto/ExamplesDto.kt
package de.viktorlevin.voicedownloader.dto

data class ExamplesDto(val result: List<Result> = emptyList()) {

    data class Result(val text: String?, val examples: List<Example> = emptyList())

    data class Example(val src: String?, val dst: String?, val ref: Reference?)

    data class Reference(val title: String?, val type: String?)
}